docs/admin/service_manager.md

```diff
@@ -0,0 +1,189 @@
+# MasterMind Service Manager
+
+## Introduction
+
+The Mastermind Service Manager is a microservice that is in charge of the
+deployment of Docker [stacks](https://docs.docker.com/get-started/part5/#prerequisites)
+to remote Docker Swarm clusters. It comprises of a [Swagger](https://swagger.io/)
+generated  API for receiving incoming requests and a custom made Python Docker
+library that leverages [docker-py](https://docker-py.readthedocs.io/en/stable/)
+and [pyyaml](http://pyyaml.org/wiki/PyYAML) in order to read and parse a
+docker-compose file, verify if there is any missing Docker component
+(e.g. Volumes and Networks) and deploy the services described in the
+docker-compose file on the target engine.
+
+This is an internal component in the MasterMind architecture and as such it is
+not visible to end users of Mastermind. More specifically, a user must send a
+request to MasterMind API which then will be forwarded to the Service Manager.
+This component is described in detail here as it may be of use in other cases;
+the docker-engine API does not provide functionality required to deploy a
+stack - that was the reason this component was developed.
+
+### High-level architecture
+
+![image alt text](../images/service_manager_architecture.png)
+
+The picture above depicts the high-level architecture of the MasterMind Service
+Manager which is comprised of three major components - [Nginx](https://www.nginx.com/),
+[Gunicorn](http://gunicorn.org/) and the [Service Manager](https://github.com/icclab/MasterMind-ServiceManager).
+As mentioned above the Service manager comprises of a Swagger API for receiving
+incoming requests, this API is based on [Flask](http://flask.pocoo.org/)
+micro-framework which handles one request at a time, and by leveraging Nginx
+and Gunicorn the overall performance of the service is increased to allow
+multiple requests to be served at a time.
+
+## Currently supported API endpoints - MasterMind Service Manager
+
+* `POST /v1/stack` - Create a new stack deploying services, creating networks,
+  volumes, secrets and configs defined in the docker-compose file.
+
+* `POST /v1/stack/{name}` - Return the status of an existing stack with the
+  services currently running in the Stack and the total amount of replicas.
+
+* `POST /v1/stack/delete/{name}` - Delete the stack removing services and
```

Comment:
I'd have left the same `v1/stack/{name}` but changing the http verb to delete. Not a change request, just sharing what I've seen most often :)

Reply:
You mean change it to a DELETE `v1/stack/{name}`? We have tried this but the content body gets dropped when using this HTTP method. Adding those parameters to the URL would make it really large.